src/components/molecules/SearchAutocomplete/index.tsx

```diff
@@ -106,34 +112,47 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
           e.preventDefault();
         }}
       >
-        <S.Label htmlFor="input-search" id="input-search-label">
-          <S.SearchIcon />
-        </S.Label>
-        <S.Input
-          id="input-search"
-          type="text"
-          autoComplete="off"
-          placeholder="Szukaj problemów matematycznych"
-          aria-label="Szukaj problemów"
-          aria-autocomplete="list"
-          aria-controls="autocomplete-options"
-          value={searchTerm}
-          onChange={handleChange}
-          onKeyDown={handleKeyDown}
-          onKeyUp={handleKeyUp}
-          onSubmit={(e) => {
-            e.preventDefault();
-          }}
-          showSuggestions={showSuggestions}
-        />
+        {searchTerm && suggestionType === "tag" ? (
```

Comment:
Here we'd have 
```suggestion
        {searchTerm && !!tag ? (
```
and `tag` would come from a parent altogether with `setTag`